plugins/inputs/ctrlx_datalayer/ctrlx_datalayer_subscription_test.go

```diff
@@ -0,0 +1,315 @@
+		PublishInterval   config.Duration
+		KeepaliveInterval config.Duration
+		ErrorInterval     config.Duration
+		SamplingInterval  config.Duration
+		QueueSize         uint
+		QueueBehaviour    string
+		DeadBandValue     float64
+		ValueChange       string
+		OutputJSONString  bool
+	}
+	type args struct {
+		id string
+	}
+	tests := []struct {
+		name     string
+		fields   fields
+		args     args
+		wantBody SubscriptionRequest
+		wantErr  bool
+	}{
+		{
+			name: "Should_Return_Expected_Request",
+			fields: fields{
+				Nodes: []Node{
+					{
+						Name:    "node1",
+						Address: "path/to/node1",
+						Tags:    map[string]string{},
+					},
+					{
+						Name:    "node2",
+						Address: "path/to/node2",
+						Tags:    map[string]string{},
+					},
+				},
+				Tags:              map[string]string{},
+				Measurement:       "",
+				PublishInterval:   config.Duration(2 * time.Second),
+				KeepaliveInterval: config.Duration(10 * time.Second),
+				ErrorInterval:     config.Duration(20 * time.Second),
+				SamplingInterval:  config.Duration(100 * time.Millisecond),
+				QueueSize:         100,
+				QueueBehaviour:    "DiscardNewest",
+				DeadBandValue:     1.12345,
+				ValueChange:       "StatusValueTimestamp",
+				OutputJSONString:  true,
+			},
+			args: args{
+				id: "sub_id",
+			},
+			wantBody: SubscriptionRequest{
+				Properties: SubscriptionProperties{
+					KeepaliveInterval: 10000,
+					Rules: []Rule{
+						{
+							"Sampling",
+							Sampling{
+								SamplingInterval: 100000,
+							},
+						},
+						{
+							"Queueing",
+							Queueing{
+								QueueSize: 100,
+								Behaviour: "DiscardNewest",
+							},
+						},
+						{
+							"DataChangeFilter",
+							DataChangeFilter{
+								DeadBandValue: 1.12345,
+							},
+						},
+						{
+							"ChangeEvents",
+							ChangeEvents{
+								ValueChange: "StatusValueTimestamp",
+							},
+						},
+					},
+					ID:              "sub_id",
+					PublishInterval: 2000,
+					ErrorInterval:   20000,
+				},
+				Nodes: []string{
+					"path/to/node1",
+					"path/to/node2",
+				},
+			},
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			s := &Subscription{
+				Nodes:             tt.fields.Nodes,
+				Tags:              tt.fields.Tags,
+				Measurement:       tt.fields.Measurement,
+				PublishInterval:   tt.fields.PublishInterval,
+				KeepaliveInterval: tt.fields.KeepaliveInterval,
+				ErrorInterval:     tt.fields.ErrorInterval,
+				SamplingInterval:  tt.fields.SamplingInterval,
+				QueueSize:         tt.fields.QueueSize,
+				QueueBehaviour:    tt.fields.QueueBehaviour,
+				DeadBandValue:     tt.fields.DeadBandValue,
+				ValueChange:       tt.fields.ValueChange,
+				OutputJSONString:  tt.fields.OutputJSONString,
+			}
+			got := s.createRequest(tt.args.id)
+			if !reflect.DeepEqual(got, tt.wantBody) {
+				t.Errorf("Subscription.createRequest() = %v, want %v", got, tt.wantBody)
+			}
```

Comment:
```suggestion
			require.Equal(t, tt.wantBody, got)
```

And the same for the other tests..